server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -430,41 +430,58 @@ public SnapshotInfo(
         Map<String, IndexSnapshotDetails> indexSnapshotDetails
     ) {
         this.snapshotId = Objects.requireNonNull(snapshotId);
-        this.indices = Collections.unmodifiableList(Objects.requireNonNull(indices));
-        this.dataStreams = Collections.unmodifiableList(Objects.requireNonNull(dataStreams));
-        this.featureStates = Collections.unmodifiableList(Objects.requireNonNull(featureStates));
+        this.indices = List.copyOf(indices);
```

Comment:
Do these changes translate to 7.x ok? I guess we'll end up using `org.elasticsearch.core.List#copyOf` which does a complete copy every time...